### PR TITLE
Python: Add routing header

### PR DESCRIPTION
**Pull request by eoogbe**
Related api_core PR: GoogleCloudPlatform/google-cloud-python#4336

### CODE REVIEW

**Comment by eoogbe** [comment]
@lukesneeringer @vam-google PTAL

File: src/main/java/com/google/api/codegen/transformer/HeaderRequestParamTransformer.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import com.google.api.codegen.config.TransportProtocol;
+import com.google.api.codegen.viewmodel.HeaderRequestParamView;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.MessageType;
+import com.google.api.tools.framework.model.Method;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+
+public class HeaderRequestParamTransformer {
+  public List<HeaderRequestParamView> generateHeaderRequestParams(MethodContext context) {
```
Comment (saicheems) [inline src/main/java/com/google/api/codegen/transformer/HeaderRequestParamTransformer.java]: These can be static.

Comment (eoogbe) [reply src/main/java/com/google/api/codegen/transformer/HeaderRequestParamTransformer.java]: They haven't been for the other transformers.

File: src/main/resources/com/google/api/codegen/py/main.snip [hunk 2]
```diff
@@ -329,3 +345,25 @@
         {@param.keyName}={@param.name}
     @end
 @end
+
+@private routingHeader(params)
+  b'{@routingHeaderFmtStr(params)}'.format({@routingHeaderFmtArgs(params)})
```
Comment (lukesneeringer) [inline src/main/resources/com/google/api/codegen/py/main.snip]: `bytes` objects do not have a `format` method in Python 3.

File: src/test/java/com/google/api/codegen/testdata/py/py_library.baseline [hunk 3]
```diff
@@ -1532,7 +1532,12 @@ class LibraryServiceClient(object):
         request = library_pb2.CreateBookRequest(
             name=name,
             book=book)
-        return self._create_book(request, retry=retry, timeout=timeout)
+
+        metadata = [
+            (b'x-google-request-params',
```
Comment (lukesneeringer) [inline src/test/java/com/google/api/codegen/testdata/py/py_library.baseline]: Same issue here. These can be UTF-8 strings, though.

**Pull request merged**